File: Data Structures & Algorithms/anagram-groups/submission-0.py
from collections import defaultdict

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        res = []

        def counting_letters(word: str) -> tuple(int):
            counts = [0] * 26

            for letter in word:
                counts[ord(letter.lower()) - ord('a')] += 1
            
            return tuple(counts)
        
        result = defaultdict(list)

        for word in strs:
            char_count = counting_letters(word)

            result[char_count].append(word)
        
        return [value for value in result.values()]
        

File: Data Structures & Algorithms/longest-common-prefix/submission-0.go
func longestCommonPrefix(strs []string) string {
    if len(strs) == 1 {
		return strs[0]
	}
	var result []byte
	for i := 0; i < len(strs[0]); i++ {
		letter := strs[0][i]
		for j := 1; j < len(strs); j++ {
			if i >= len(strs[j]) {
				return string(result)
			}
			if letter != strs[j][i] {
				return string(result)
			}
		}
		result = append(result, letter)
	}
	return string(result )
}
